src/plugins/unified_histogram/public/types.ts

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { Theme } from '@kbn/charts-plugin/public/plugin';
+import type { IUiSettingsClient } from '@kbn/core/public';
+import type { DataPublicPluginStart } from '@kbn/data-plugin/public';
+import type { FieldFormatsStart } from '@kbn/field-formats-plugin/public';
+import type { Duration, Moment } from 'moment';
+import type { Unit } from '@kbn/datemath';
+import type { SerializedFieldFormat } from '@kbn/field-formats-plugin/common';
+
+/**
+ * The fetch status of a unified histogram request
+ */
+export type UnifiedHistogramFetchStatus =
+  | 'uninitialized'
+  | 'loading'
+  | 'partial'
+  | 'complete'
+  | 'error';
+
+/**
+ * The services required by the unified histogram components
+ */
+export interface UnifiedHistogramServices {
+  data: DataPublicPluginStart;
+  theme: Theme;
+  uiSettings: IUiSettingsClient;
+  fieldFormats: FieldFormatsStart;
+}
+
+interface Column {
+  id: string;
+  name: string;
+}
+
+interface Row {
+  [key: string]: number | 'NaN';
+}
+
+interface Dimension {
+  accessor: 0 | 1;
+  format: SerializedFieldFormat<{ pattern: string }>;
+  label: string;
+}
+
+interface Ordered {
+  date: true;
+  interval: Duration;
+  intervalESUnit: string;
+  intervalESValue: number;
+  min: Moment;
+  max: Moment;
+}
+
+interface HistogramParams {
+  date: true;
+  interval: Duration;
+  intervalESValue: number;
+  intervalESUnit: Unit;
+  format: string;
+  bounds: HistogramParamsBounds;
+}
+
+export interface HistogramParamsBounds {
+  min: Moment;
+  max: Moment;
+}
+
+export interface Table {
+  columns: Column[];
+  rows: Row[];
+}
+
+export interface Dimensions {
+  x: Dimension & { params: HistogramParams };
+  y: Dimension;
+}
+
+// @ts-ignore
+// eslint-disable-next-line @typescript-eslint/naming-convention
+type buildChartData = typeof import('./chart/build_chart_data').buildChartData;
+
+/**
+ * The chartData object returned by {@link buildChartData} that
+ * should be used to set {@link UnifiedHistogramChartContext.data}
```

Comment:
It's great that you take care of documentation, and provide links in the description. I also should do this more often 😅 . in this case I'm interested. Doesn't it work without having to import   `buildChartData` as a type, that's not in use in the file, except in the documentation part. and also having to add a _@ts-ignore_ ? removed it and it works form me in IDEA, can click on the link, I'm jumping to the right file.

One other first observation, DataDiscover needs to claim code ownership of this 💎  (before others do so)

Reply:
I wasn't able to get it working any other way in VS Code except this or a regular import statement, but our eslint won't let us ignore unused import statements so I went with this. Can you post a screenshot of how it looks in IDEA since I'm curious now? 😄 Either way I'm ok to remove this line since it sounds like it's probably a VS Code issue.

Also good catch, I'll update the `CODEOWNERS` file to add us.

Reply:
It looks like IDEA is smarter than VS Code 🙃 I wonder how it knows which file to navigate to without the import. In any case I can remove these imports and let the IDEs figure it out instead since it's a pretty minor thing. It's possible VS Code will improve support for `@link`s in the future anyway.

Reply:
I ended up removing these imports.